lib/runtime/adapter-read.js

```diff
@@ -114,6 +115,15 @@ class AdapterRead {
         return errors.runtimeError(code, info);
     }
 
+    // Helper to obtain a throttled function to emit warnings.
+    throttledWarn(code, info, interval) {
+        interval = interval || 1000;
+        info = info || {};
+        return _.throttle(() => {
+            this.trigger('warning', this.runtimeError(code, info));
+        }, interval);
+    }
```

Comment:
@bkutil That seems like it could be a useful follow-up to consolidate even further.

Regardless I think it would still make sense to have a `this.throttledWarning` utility on the adapter class itself, so I'm going to merge this as-is and then we can separately consider whether to refactor further.